### PR TITLE
add delete sync block and test case

### PR DESCRIPTION
**Pull request by jackzhhuang**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

### CODE REVIEW

File: storage/src/block/mod.rs [hunk 1]
```diff
@@ -408,6 +408,10 @@ impl BlockStorage {
         self.dag_sync_block_storage.remove(block_id)
     }
 
+    pub fn delete_all_dag_sync_blocks(&self) -> Result<()> {
```
Comment (nkysg) [inline storage/src/block/mod.rs:411]: I only see you call this function in test code? Does this function use in sync process?

File: storage/src/block/mod.rs [hunk 2]
```diff
@@ -408,6 +408,10 @@ impl BlockStorage {
         self.dag_sync_block_storage.remove(block_id)
```
Comment (nkysg) [inline storage/src/block/mod.rs:408]: I see you have implemented this function. I don't know where do you want to call  this function. It likes garbarge collection produce. If your don't call this function, the dag_sync_block storage grows infinitely.

File: storage/src/block/mod.rs (hunk 1 above)
Comment (jackzhhuang) [reply storage/src/block/mod.rs:411]: Please see: sync/src/tasks/block_sync_task.rs line 761

File: sync/src/tasks/block_sync_task.rs [hunk 3]
```diff
@@ -513,13 +501,8 @@ where
                             process_dag_ancestors
                                 .insert(block.header().id(), block.header().clone());
 
-                            // execute the children if their parents are ready
-                            // if all children are executed, delete it.
                             self.execute_if_parent_ready(executed_block.block.id())?;
 
-                            // todo: how to delete the block?
-                            // self.local_store.delete_dag_sync_block(executed_block.block.id())?;
```
Comment (nkysg) [inline sync/src/tasks/block_sync_task.rs:521]: It seems that you don't think clearly how to process usefuless dag_sync_block.You can't remove this comment.

File: storage/src/block/mod.rs (hunk 1 above)
Comment (nkysg) [reply storage/src/block/mod.rs:411]: Does delete_dag_sync_block this function still use in some place?

**Comment by nkysg** [comment]
Please use rebase not merge.

File: storage/src/block/mod.rs (hunk 1 above)
Comment (jackzhhuang) [reply storage/src/block/mod.rs:411]: Yes, see the new branch: https://github.com/starcoinorg/starcoin/pull/4120

**Pull request closed without merging**